app/src/main/java/com/youssef/fixit/Auth/LoginFragment/LoginPresenter.java

```diff
@@ -54,9 +55,9 @@ public void onResponse(Call<Register> call, Response<Register> response) {
                         SplashScreen.MyRole = roles.get(0);
                         Log.d(TAG, response.body().getData().getId() + "");
                         Log.d(TAG, SplashScreen.MyToken);
-                        view.saveString("token", data.getToken());
-                        view.saveString("role", data.getRoles().get(0));
-                        view.saveInt("my_id", data.getId());
+                        iSharedPreference.save("token", data.getToken());
+                        iSharedPreference.save("role", data.getRoles().get(0));
+                        iSharedPreference.save("my_id", data.getId());
                         view.onLoginSuccessful();
```

Comment:
Since we are releasing the view on the `presenter#destroy` it's better to check if the view is not null before accessing it.